### PR TITLE
Add some internal documentation for flake support objects.

### PR DESCRIPTION
**Pull request by kquick**
Please feel free to update any parts that are not correct or poorly stated, but hopefully this will help for comprehension and maintenance of the flake support.

### CODE REVIEW

File: src/libfetchers/fetchers.hh [hunk 1]
```diff
@@ -21,6 +21,13 @@ struct Tree
 
 struct InputScheme;
 
+// The Input object is generated by a specific fetcher, based on the
+// user-supplied input attribute in the flake.nix file, and contians
```
Comment (edolstra) [inline src/libfetchers/fetchers.hh:25]:
```suggestion
// user-supplied input attribute in the flake.nix file, and contains
```

File: src/libexpr/flake/flakeref.hh [hunk 2]
```diff
@@ -12,10 +12,25 @@ class Store;
 
 typedef std::string FlakeId;
 
+// The FlakeRef represents a local nix store reference to a flake
```
Comment (edolstra) [inline src/libexpr/flake/flakeref.hh:15]: The "local nix store reference" bit is confusing. Maybe something like "A flake reference specifies how to fetch a flake (e.g. from a Git repository). It is created from a URL-like syntax (e.g. 'github:NixOS/patchelf') or from an attrset representation (e.g. '{ type = "github"; owner = "NixOS"; repo = "patchelf"; }').".

File: src/libexpr/flake/flake.hh [hunk 3]
```diff
@@ -17,20 +17,41 @@ struct FlakeInput;
 
 typedef std::map<FlakeId, FlakeInput> FlakeInputs;
 
+// FlakeInput is the flake-level parsed form of the "input" entries in
```
Comment (edolstra) [inline src/libexpr/flake/flake.hh:20]: Could you use `/* ... */` style comments? We mostly use `//` for single-line comments.

File: src/libexpr/flake/flakeref.hh (hunk 2 above)
Comment (kquick) [reply src/libexpr/flake/flakeref.hh]: Thanks, that does sound clearer.  I made a couple of additional adjustments to this description that occurred to me as I was working on it; let me know what you think of the updated version.

File: src/libfetchers/fetchers.hh (hunk 1 above)
Comment (kquick) [reply src/libfetchers/fetchers.hh]: Thanks!

File: src/libexpr/flake/flake.hh (hunk 3 above)
Comment (kquick) [reply src/libexpr/flake/flake.hh]: Sure.  I made this in its own commit with no other wording changes (although some `fmt` wrapping updates occurred); all wording changes are in different commits to make it easier to see those changes.

**Comment by kquick** [comment]
Thanks for the suggestions, @edolstra .  I've made all the updates and this is ready for reviewing again.

**Comment by kquick** [comment]
@edolstra  pinging you since it would great to get this merged before things diverged.

**Pull request merged**